From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Nam Le <namle888029@gmail.com>
Date: Tue, 30 Dec 2025 10:47:40 +0700
Subject: [PATCH] 8837: Symbolic Propagator: fix duplicated ref creation
 attempt in applyPCode() for CALLIND Pcode

fix check all references when validating for replacement

clean up unnecessary whitespace

cleanup whitespace

Clean up logic
---
 .../ghidra/program/util/SymbolicPropogator.java    | 14 +++++++++++---
 1 file changed, 11 insertions(+), 3 deletions(-)

diff --git a/Ghidra/Features/Base/src/main/java/ghidra/program/util/SymbolicPropogator.java b/Ghidra/Features/Base/src/main/java/ghidra/program/util/SymbolicPropogator.java
index 4539934980..391663e6a4 100644
--- a/Ghidra/Features/Base/src/main/java/ghidra/program/util/SymbolicPropogator.java
+++ b/Ghidra/Features/Base/src/main/java/ghidra/program/util/SymbolicPropogator.java
@@ -1002,10 +1002,18 @@ public class SymbolicPropogator {
 									// if the value didn't get changed, then the real value isn't in here, don't make a reference
 									if (target != null) {
 										Reference[] refs = instruction.getReferencesFrom();
+
+										boolean refExisted = false;
+										for (Reference ref: refs) {
+											if (ref.getToAddress().equals(target)) {
+												refExisted = true;
+												break;
+											}
+										}
+
 										// make sure we aren't replacing a read ref with a call to the same place
-										if (refs.length <= 0 ||
-											!refs[0].getToAddress().equals(target)) {
-	
+										if (!refExisted) {
+
 											target = makeReference(vContext, instruction, Reference.MNEMONIC,
 												//  Use target in case location has shifted (external...)
 												target.getAddressSpace().getSpaceID(),
-- 
2.45.1

